### PR TITLE
OSOE-64: Add samples project to Orchard-Vue.js ⭐

### PR DESCRIPTION
**Pull request by sarahelsaig**
OSOE-64

### CODE REVIEW

**Review by barthamark — APPROVED** [review]

File: test/Lombiq.OSOCE.Tests.UI/Tests/BehaviorVueTests.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+        [Theory, Chrome]
+        public Task RecipeDataShouldBeDisplayedCorrectly(Browser browser) =>
+            ExecuteTestAfterSetupAsync(
+                async context =>
+                {
+                    // Disabling some bothersome features that make relevant testing harder.
+                    await context.SignInDirectlyAsync();
+                    await context.DisableFeatureDirectlyAsync("Lombiq.Privacy.ConsentBanner");
+                    await context.DisableFeatureDirectlyAsync("Lombiq.Hosting.Azure.ApplicationInsights");
```
Comment (Piedone) [inline test/Lombiq.OSOCE.Tests.UI/Tests/BehaviorVueTests.cs]: AI shouldn't affect anything. What's the issue?

Comment (sarahelsaig) [reply test/Lombiq.OSOCE.Tests.UI/Tests/BehaviorVueTests.cs]: The same thing as last time, trying to access the "https://dc.services.visualstudio.com/v2/track". The thing that we thought switching to offline mode fixes for good. However [AI is already in offline mode in the project](https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/blob/issue/OSOE-64/test/Lombiq.OSOCE.Tests.UI/UITestBase.cs#L45) so apparently that's not enough.

Comment (Piedone) [reply test/Lombiq.OSOCE.Tests.UI/Tests/BehaviorVueTests.cs]: At that time switching to offline mode fixed this (as it should), if this reappeared then please look into it because it breaks UI testing as a whole.

Comment (sarahelsaig) [reply test/Lombiq.OSOCE.Tests.UI/Tests/BehaviorVueTests.cs]: I have looked into it and found nothing, hence the DisableFeatureDirectlyAsync usage.

Comment (Piedone) [reply test/Lombiq.OSOCE.Tests.UI/Tests/BehaviorVueTests.cs]: Can you reproduce?

Comment (sarahelsaig) [reply test/Lombiq.OSOCE.Tests.UI/Tests/BehaviorVueTests.cs]: Nope, I couldn't repro locally. Which made the whole investigation super frustrating and sluggish.

Comment (sarahelsaig) [reply test/Lombiq.OSOCE.Tests.UI/Tests/BehaviorVueTests.cs]: Why do we need the module enabled in the tests anyway? I don't see any value in enabling the feature just to immediately gut its functionality (but apparently not completely).

File: test/Lombiq.OSOCE.Tests.UI/Tests/BehaviorVueTests.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    [Theory, Chrome]
+    public Task RecipeDataShouldBeDisplayedCorrectly(Browser browser) =>
+        ExecuteTestAfterSetupAsync(
+            async context =>
+            {
+                // Disabling some bothersome features that make relevant testing harder.
+                await context.SignInDirectlyAsync();
+                await context.DisableFeatureDirectlyAsync("Lombiq.Privacy.ConsentBanner");
```
Comment (Piedone) [inline test/Lombiq.OSOCE.Tests.UI/Tests/BehaviorVueTests.cs]: Thinking about it, instead of enabling `Lombiq.Privacy` in the test recipe, and then disabling it here and in Lombiq.OSOCE.BasicOrchardFeaturesTests.recipe, as well as `BehaviorFileUploadTests` and Lombiq.OSOCE.NuGet.BasicOrchardFeaturesTests.recipe of the NuGet solution, we could just enable it in a single test in Lombiq.OSOCE.Tests.UI. It causes too many issues for other features.

File: test/Lombiq.OSOCE.Tests.UI/Tests/BehaviorVueTests.cs (hunk 1 above)
Comment (Piedone) [reply test/Lombiq.OSOCE.Tests.UI/Tests/BehaviorVueTests.cs]: It's an example of testing an otherwise online feature offline (`ApplicationInsightsTrackingShouldBePresent`), while demonstrating passing configuration to the app.

If it's not in offline mode then that's a big issue because that means that the configuration arguments are not reaching the app.

File: test/Lombiq.OSOCE.Tests.UI/Tests/BehaviorVueTests.cs (hunk 2 above)
Comment (sarahelsaig) [reply test/Lombiq.OSOCE.Tests.UI/Tests/BehaviorVueTests.cs]: On a semi-related note: Looking at the `ShellFeaturesController` in `Lombiq.Tests.UI.Shortcuts`, do you see any reason why it shouldn't be `[AllowAnonymous]` to enable/disable features in the tests without having to sign in first? (It's already `[DevelopmentAndLocalhostOnly]` so I don't think there is a security concern.)

File: test/Lombiq.OSOCE.Tests.UI/Tests/BehaviorVueTests.cs (hunk 1 above)
Comment (sarahelsaig) [reply test/Lombiq.OSOCE.Tests.UI/Tests/BehaviorVueTests.cs]: This rings a bell. Recently I have found a bug in the argument building and [fixed it here](https://github.com/Lombiq/UI-Testing-Toolbox/commit/e70011d658f26249a623c683da208dc0e17ee7a5) while working on the Linux port. It didn't break anything visibly on Windows, but totally freaked out on Linux. Maybe it had more subtle problems on Windows, including this bug? I will cherry pick it to this branch and then we'll see.

Comment (sarahelsaig) [reply test/Lombiq.OSOCE.Tests.UI/Tests/BehaviorVueTests.cs]: Yep, looks like that did the trick. I never would've figured this out if not for the (admittedly painful) work in  OSOE-49...

Comment (Piedone) [reply test/Lombiq.OSOCE.Tests.UI/Tests/BehaviorVueTests.cs]: Wow, nice catch!

File: test/Lombiq.OSOCE.Tests.UI/Tests/BehaviorVueTests.cs (hunk 2 above)
Comment (Piedone) [reply test/Lombiq.OSOCE.Tests.UI/Tests/BehaviorVueTests.cs]: It's OK to have it with anonymous access, just as the others, but I don't know why it wouldn't work like that already, since there's no authentication on it.

File: test/Lombiq.OSOCE.Tests.UI/Tests/BehaviorVueTests.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        ExecuteTestAfterSetupAsync(
+            async context =>
+            {
+                await context.SignInDirectlyAsync();
+                await context.TestVueSampleBehaviorAsync();
+            },
+            browser);
+
+    [Theory, Chrome]
+    public Task ConsentBannerShouldWork(Browser browser) =>
```
Comment (Piedone) [inline test/Lombiq.OSOCE.Tests.UI/Tests/BehaviorVueTests.cs:30]: This is not related to Vue.

File: src/Lombiq.OSOCE.Web/Recipes/Lombiq.OSOCE.Tests.recipe.json [hunk 4]
```diff
@@ -32,12 +32,13 @@
         "Lombiq.Hosting.Tenants.Management",
         "Lombiq.JsonEditor",
         "Lombiq.OSOCE.Samples",
-        "Lombiq.Privacy.ConsentBanner",
+        // "Lombiq.Privacy.ConsentBanner", // We only enable this on a specific test, because it overlaps with too many things.
```
Comment (Piedone) [inline src/Lombiq.OSOCE.Web/Recipes/Lombiq.OSOCE.Tests.recipe.json:35]: This is not fully addressed: https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/pull/46#discussion_r829566782

Comment (sarahelsaig) [reply src/Lombiq.OSOCE.Web/Recipes/Lombiq.OSOCE.Tests.recipe.json:35]: ngl that comment made my head spin. It also caught me off guard that some of these files are not actually in the solution - I mean it's not surprising for the recipe, but why is the `BehaviorFileUploadTests` NuGet only?

Comment (Piedone) [reply src/Lombiq.OSOCE.Web/Recipes/Lombiq.OSOCE.Tests.recipe.json:35]: They are all in the solution, but there are two solutions, the other being the NuGetTest folder.

Comment (sarahelsaig) [reply src/Lombiq.OSOCE.Web/Recipes/Lombiq.OSOCE.Tests.recipe.json:35]: They are all in *a solution. Not the default one tho. :)
Can't these files be referenced from the root solution somehow? This hurts IDE discoverability.

Comment (Piedone) [reply src/Lombiq.OSOCE.Web/Recipes/Lombiq.OSOCE.Tests.recipe.json:35]: I suppose you can reference the NuGet recipe from the root solution, though it'd be confusing. Same with the projects unless they somehow clash DLLs (NuGet packages vs local projects). We've evaluated having those projects in the same solution vs in a different one (vs in a different repo), you can check out why it's like it is now from here: https://lombiq.atlassian.net/browse/INFRA-92?focusedCommentId=61703
